gematria/datasets/process_and_filter_bbs_lib.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "gematria/llvm/llvm_architecture_support.h"
+
+namespace gematria {
+
+class BBProcessorFilter {
```

Comment:
Please add a explaining what the class is for, and also a comment for non-trivial members (right now, probably just `processBasicBlock`).